hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java

```diff
@@ -306,4 +311,37 @@ private void rejoinJob(Throwable t) {
             future.internalCompleteExceptionally(new CancellationException(msg));
         }
     }
+
+    static final class JobMetricsImpl implements JobMetrics {
+
+        private final Map<String, Long> metrics;
+
+        private JobMetricsImpl(Map<String, Long> metrics) {
+            this.metrics = Collections.unmodifiableMap(new HashMap<>(metrics));
+        }
+
+        static JobMetrics of(Map<String, Long> metrics) {
+            return new JobMetricsImpl(metrics);
+        }
+
+        @Nonnull
+        @Override
+        public Collection<String> getMetricNames() {
+            return metrics.keySet();
+        }
+
+        @Override
+        public Long getMetricValue(String name) throws IllegalArgumentException {
+            if (name == null) {
+                throw new IllegalArgumentException("Metric name can't be null!");
```

Comment:
Don't use exclamation marks for exceptions!! :)

Reply:
Ok, fixed.